nixos/modules/services/web-apps/nextcloud.nix

```
@@ -899,8 +874,8 @@ in {
             mkExport = { arg, value }: "export ${arg}=${value}";
             dbpass = {
               arg = "DBPASS";
-              value = if c.dbpassFile != null
-                then ''"$(<"${toString c.dbpassFile}")"''
+              value = if cfg.secretFile != null
+                then ''"$(${lib.getExe pkgs.jq} -r '.dbpass' "${toString cfg.secretFile}")"''
```

This will cause problems when `secretFile` is used but doesn't contain a `dbpass` since `jq ".doesNotExist"` will return the literal string `null`. Setting a default string as a fallback should fix this.

```suggestion
              value = if cfg.secretFile != null
                then ''"$(${lib.getExe pkgs.jq} -r '.dbpass // \"\"' "${toString cfg.secretFile}")"''
```